lib/core/src/sdk.rs

```diff
@@ -1313,73 +1314,74 @@ impl LiquidSdk {
                         (asset_id, receiver_amount_sat, Some(fees_sat), None)
                     }
                     PayAmount::Asset {
-                        asset_id,
+                        to_asset,
+                        from_asset,
                         receiver_amount,
                         estimate_asset_fees,
-                        pay_with_bitcoin,
                     } => {
-                        let estimate_asset_fees = estimate_asset_fees.unwrap_or(false);
-                        let asset_metadata = self.persister.get_asset_metadata(&asset_id)?.ok_or(
-                            PaymentError::AssetError {
-                                err: format!("Asset {asset_id} is not supported"),
-                            },
-                        )?;
-                        let receiver_amount_sat = asset_metadata.amount_to_sat(receiver_amount);
+                        let from_asset = from_asset.unwrap_or(to_asset.clone());
+                        let _payer_asset_metadata = self
+                            .persister
+                            .get_asset_metadata(&from_asset)?
+                            .ok_or(PaymentError::AssetError {
+                                err: format!("Asset {from_asset} is not supported"),
+                            })?;
+                        let receiver_asset_metadata = self
+                            .persister
+                            .get_asset_metadata(&to_asset)?
+                            .ok_or(PaymentError::AssetError {
+                                err: format!("Asset {to_asset} is not supported"),
+                            })?;
+                        let receiver_amount_sat =
+                            receiver_asset_metadata.amount_to_sat(receiver_amount);
+
+                        let asset_fees = if estimate_asset_fees.unwrap_or(false) {
+                            self.payjoin_service
+                                .estimate_payjoin_tx_fee(&from_asset, receiver_amount_sat)
```

Comment:
Shouldn't it be `to_asset` here?

```suggestion
                                .estimate_payjoin_tx_fee(&to_asset, receiver_amount_sat)
```

Reply:
You're right. I had set it to `from` because the first argument actually refers to the fee asset, so it would make sense. But later on we build the tx with the `to_asset`, so this needs correction.

Reply:
Same as [comment](https://github.com/breez/breez-sdk-liquid/pull/986#discussion_r2262990890), they are now mutually exclusive.